src/main/java/cs/ualberta/cmput402/tictactoe/TicTacToeGame.java

```diff
@@ -31,22 +34,48 @@ public void promptNextPlayer(){
 
     public void playGame(){
         Scanner keyboardScanner = new Scanner(System.in);
+        boolean shouldContinue = true;
+        
+        while (shouldContinue) {
+	        while (board.getWinner() == null && board.maxTurns() == false){
+	            board.printBoard();
+	            promptNextPlayer();
+	            String line = keyboardScanner.nextLine();
+	            String input[] = line.split(",");
+	            try {
+	                board.playMove(Integer.parseInt(input[0]), Integer.parseInt(input[1]));
+	            } catch (InvalidMoveException e) {
+	                System.out.println("Invalid coordinates. Try again");
+	                promptNextPlayer();
+	            }
+	        }
 
-        while (board.getWinner() == null){
-            board.printBoard();
-            promptNextPlayer();
-            String line = keyboardScanner.nextLine();
-            String input[] = line.split(",");
-            try {
-                board.playMove(Integer.parseInt(input[0]), Integer.parseInt(input[1]));
-            } catch (InvalidMoveException e) {
-                System.out.println("Invalid coordinates. Try again");
-                promptNextPlayer();
-            }
+	        board.printBoard();
+	        if (board.getWinner() != null) {
+	        	System.out.println("Player " + board.getWinner() + " has won the game!");	
+	        	
+	        	if(Player.X == board.getWinner()) {
+	        		scoreBoard.setXwin();
+	        	} else if (Player.O == board.getWinner()) {
+	        		scoreBoard.setOwin();
+	        	}
+	        }
+	        else {
+	        	System.out.println("The game has ended in a tie.");
+	        	scoreBoard.setTie();
+	        }
+	        scoreBoard.printScoreBoard();
+	        System.out.println("Do you want to play another game? (Types yes to play or anything else to exit)");
+	        String userInput = keyboardScanner.nextLine();
+	        
+	        if(userInput.equals("yes")) {
+	        	board.resetGame();
```

Comment:
Although this may indeed work we do not yet see any issue with reusing the functions provided in the starter code, we think that using the functions defined in the base program is a better practice for the purposes of this project.

Reply:
Your approach does not have a problem. The only thing is that, with your approach, there is a higher probability that you make a mistake. (As you forgot to initialize one of the variables.)
With the other approach, there is a lower probability that a mistake occurs. That's all.